src-tauri/src/config.rs

```diff
@@ -439,4 +455,70 @@ impl LauncherConfig {
       }
     }
   }
+
+  pub fn game_enabled_textured_packs(&self, game_name: &String) -> Vec<String> {
+    match SupportedGame::from_str(game_name) {
+      Ok(game) => {
+        // Retrieve relevant game from config
+        match self.games.get(&game) {
+          Some(game) => match &game.features {
+            Some(features) => features.texture_packs.to_owned(),
+            None => Vec::new(),
+          },
+          None => {
+            log::warn!(
+              "Could not find game to check which texture packs are enabled: {}",
+              game_name
+            );
+            Vec::new()
+          }
+        }
+      }
+      Err(_) => {
+        log::warn!(
+          "Could not find game to check which texture packs are enabled: {}",
+          game_name
+        );
+        Vec::new()
+      }
+    }
+  }
+
+  pub fn cleanup_game_enabled_texture_packs(
+    &mut self,
+    game_name: &String,
+    cleanup_list: Vec<String>,
+  ) -> Result<(), ConfigError> {
+    if !cleanup_list.is_empty() {
+      return Ok(());
+    }
+    match SupportedGame::from_str(game_name) {
```

Comment:
I think it would be more idiomatic to nest this less, maybe like this:

```rust
    let game = match SupportedGame::from_str(game_name) {
      Err(_) => {
        log::warn!(
          "Could not find game to check which texture packs are enabled: {}",
          game_name
        );
        return Ok(());
      },
      Ok(game) => game,
    };
    let game = match self.games.get_mut(&game) {
      None => {
        log::warn!(
          "Could not find game to check which texture packs are enabled: {}",
          game_name
        );
        return Ok(());
      },
      Some(game) => game,
    };

    if let Some(features) = &mut game.features {
      features
        .texture_packs
        .retain(|pack| cleanup_list.contains(pack));
      self.save_config()?;
    }
```

Side note: it seems there are a few places where we have to do both `SupportedGame::from_str` immediately followed by `self.games.get...`. If the map self games is supposed to have all supported games, there is no need to handle cases where from_str succeeds but the map lookup doesn't, so this module could have a helper (well, I think two are needed because of the mut):

```rust
  fn get_supported_game_config(&self, game_name: &String) -> Option<&GameConfig> {
    let game = match SupportedGame::from_str(game_name) {
      Err(_) => {
        log::warn!("Game is not supported: {}", game_name);
        return None;
      }
      Ok(game) => game,
    };
    match self.games.get(&game) {
      None => {
        log::error!("Supported game missing from games map: {}", game_name);
        None
      }
      Some(cfg) => Some(cfg),
    }
  }

  fn get_supported_game_config_mut(&mut self, game_name: &String) -> Option<&mut GameConfig> {
    let game = match SupportedGame::from_str(game_name) {
      Err(_) => {
        log::warn!("Game is not supported: {}", game_name);
        return None;
      }
      Ok(game) => game,
    };
    match self.games.get_mut(&game) {
      None => {
        log::error!("Supported game missing from games map: {}", game_name);
        None
      }
      Some(cfg) => Some(cfg),
    }
  }
```

This makes for example `game_install_version_folder` like this:
```rust
  pub fn game_install_version_folder(&self, game_name: &String) -> String {
    self
      .get_supported_game_config(game_name)
      .and_then(|cfg| cfg.version_folder.clone())
      .unwrap_or("".to_string())
  }
```

But .... perhaps `get_supported_game_config` and `get_supported_game_config_mut` could also return `Result` instead of `Option`, if the strings passed are always only _really_ supposed to be supported games.